src/compute/src/sink/persist_sink.rs

```diff
@@ -720,11 +838,29 @@ where
                                     persist_frontier
                                 );
                             }
-                            correction.with_correction_buffer(sink_metrics, sink_worker_metrics, |buffer| buffer.append(&mut data));
+
+                            if let Some(stash) = &mut desired_stash {
+                                stash.insert(data);
+                            } else {
+                                correction.with_correction_buffer(
+                                    sink_metrics,
+                                    sink_worker_metrics,
+                                    |buffer| buffer.append(&mut data),
+                                );
+                            }
 
                             continue;
                         }
                         Event::Progress(frontier) => {
+                            // Extract desired rows as positive contributions to `correction`.
+                            if let Some(stash) = &mut desired_stash {
+                                correction.with_correction_buffer(
+                                    sink_metrics,
+                                    sink_worker_metrics,
+                                    |buffer| buffer.extend(stash.drain_before(&frontier)),
```

Comment:
Uncertain if this is correct: We might be able to advance the times to the meet of persist and desired frontier to enable more consolidation immediately. I think we should leave it as-is for the moment but note it down as a path towards further optimization, unless this information just isn't available here.